services/pull/check.go

```diff
@@ -301,12 +301,23 @@ func manuallyMerged(ctx context.Context, pr *issues_model.PullRequest) bool {
 	pr.Merger = merger
 	pr.MergerID = merger.ID
 
+	ctx, committer, err := db.TxContext(ctx)
+	if err != nil {
+		log.Error("%-v db.TxContext: %v", pr, err)
+		return false
+	}
+	defer committer.Close()
 	if merged, err := pr.SetMerged(ctx); err != nil {
 		log.Error("%-v setMerged : %v", pr, err)
 		return false
 	} else if !merged {
 		return false
 	}
+	if err := committer.Commit(); err != nil {
+		log.Error("%-v committer.Commit: %v", pr, err)
+		return false
+	}
+	committer.Close()
 
 	notify_service.MergePullRequest(ctx, merger, pr)
```

Comment:
It is wrong again, the `ctx` is polluted.

No idea why you always insist to use the old ` db.TxContext(ctx)` but not new `WithTx`, always bugs.

Reply:
d483c76acc99e60989b0c1b2e0c9ae83f169da78